### PR TITLE
tests: Add ray query gpu av tests

### PR DESCRIPTION
**Pull request by arno-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7500

### CODE REVIEW

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+        #version 460
+        #extension GL_EXT_ray_query : require
+        
+        layout(set = 0, binding = 0) uniform accelerationStructureEXT tlas;
+
+        void main() {
+            rayQueryEXT query;
+            rayQueryInitializeEXT(query, tlas, gl_RayFlagsTerminateOnFirstHitEXT, 0xff, vec3(0), 0.1, vec3(0,0,1), 1000.0);
+            rayQueryProceedEXT(query);
+            // uint intersection_type = rayQueryGetIntersectionTypeEXT(query, true);
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:48]: remove (from all shaders in this file)

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+ */
+
+#include "../framework/layer_validation_tests.h"
+#include "../framework/pipeline_helper.h"
+#include "../framework/descriptor_helper.h"
+#include "../framework/gpu_av_helper.h"
+#include "../framework/ray_tracing_objects.h"
+#include "../../layers/gpu_shaders/gpu_shaders_constants.h"
+
+TEST_F(PositiveGpuAV, RayQueryComputeBasic) {
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:25]: For all tests in this file 

```suggestion
TEST_F(PositiveGpuAVRayQuery, ComputeBasic) {
```

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+    pipeline.CreateComputePipeline();
+
+    // Add TLAS binding
+    vkt::as::BuildGeometryInfoKHR tlas = vkt::as::blueprint::BuildOnDeviceTopLevel(*m_device, *m_commandBuffer);
+    pipeline.descriptor_set_->WriteDescriptorAccelStruct(0, 1, &tlas.GetDstAS()->handle());
+
+    // Add uniform buffer binding
+    vkt::Buffer uniform_buffer(*m_device, 4096, VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
+                               VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
+    pipeline.descriptor_set_->WriteDescriptorBufferInfo(1, uniform_buffer, 0, uniform_buffer.create_info().size);
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:121]: trick for all these  `WriteDescriptorBufferInfo`
```suggestion
    pipeline.descriptor_set_->WriteDescriptorBufferInfo(1, uniform_buffer, 0, VK_WHOLE_SIZE);
```

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+    m_device->wait();
+
+    // Ray query with both t_min and t_max dynamically set to 42
+    {
+        auto uniform_buffer_ptr = static_cast<float *>(uniform_buffer.memory().map());
+        uniform_buffer_ptr[0] = 42.0f;  // t_min
+        uniform_buffer_ptr[1] = 42.0f;  // t_max
+        uniform_buffer.memory().unmap();
+    }
+    m_commandBuffer->begin();
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:149]: you should be able to just resubmit, no need to rebuild a command buffer, it already has the uniform memory bound

you also forgot `pipeline.descriptor_set_->UpdateDescriptorSets();`

```c++
    m_commandBuffer->QueueCommandBuffer();
    m_device->wait();

    {
        auto uniform_buffer_ptr = static_cast<float *>(uniform_buffer.memory().map());
        uniform_buffer_ptr[0] = 42.0f;  // t_min
        uniform_buffer_ptr[1] = 42.0f;  // t_max
        uniform_buffer.memory().unmap();
    }

pipeline.descriptor_set_->UpdateDescriptorSets();
    m_commandBuffer->QueueCommandBuffer();
    m_device->wait();
```

**Review by spencer-lunarg — APPROVED** [review]
LGTM - Some small name nits

File: tests/unit/gpu_av_ray_query_positive.cpp (hunk 4 above)
Comment (arno-lunarg) [reply tests/unit/gpu_av_ray_query_positive.cpp:149]: Why would I need to call `UpdateDescriptorSet` again? Updating mapped data should not change it?

Comment (spencer-lunarg) [reply tests/unit/gpu_av_ray_query_positive.cpp]: aww, forgot it is mapped memory, ya should be good without `UpdateDescriptorSets`... but still remove the extra command buffer building

**Comment by spencer-lunarg** [comment]
@arno-lunarg  

seems this crashes on `PositiveGpuAVRayQuery.RayQueryComputeBasic` for the Galaxy s23, if you look at the crash dump you see it is crashing on `gpuav::Validator::UpdateBDABuffer`

**Comment by spencer-lunarg** [comment]
@arno-lunarg ok, so the Galaxy 23 doesn't support `shaderInt64` and seems that we use that for `buffer_device_address_enabled` to call `vmaCreateBuffer` but we never check `buffer_device_address_enabled`  in the queue submit

**Comment by arno-lunarg** [comment]
Oh nice catch

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+ */
+
+#include "../framework/layer_validation_tests.h"
+#include "../framework/pipeline_helper.h"
+#include "../framework/descriptor_helper.h"
+#include "../framework/gpu_av_helper.h"
+#include "../framework/ray_tracing_objects.h"
+#include "../../layers/gpu_shaders/gpu_shaders_constants.h"
+
+TEST_F(PositiveGpuAVRayQuery, RayQueryComputeBasic) {
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:25]:
```suggestion
TEST_F(PositiveGpuAVRayQuery, ComputeBasic) {
```

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+    vk::CmdBindDescriptorSets(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_COMPUTE, pipeline.pipeline_layout_.handle(), 0, 1,
+                              &pipeline.descriptor_set_->set_, 0, nullptr);
+    vk::CmdDispatch(m_commandBuffer->handle(), 1, 1, 1);
+    m_commandBuffer->end();
+
+    m_commandBuffer->QueueCommandBuffer();
+    m_device->wait();
+}
+
+TEST_F(PositiveGpuAVRayQuery, RayQueryComputeDynamicTminTmax) {
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:73]:
```suggestion
TEST_F(PositiveGpuAVRayQuery, ComputeDynamicTminTmax) {
```

**Comment by spencer-lunarg** [comment]
@arno-lunarg please update the test name not to duplicate `RayQuery` and ready to go

File: tests/unit/gpu_av_ray_query_positive.cpp (hunk 5 above)
Comment (arno-lunarg) [reply tests/unit/gpu_av_ray_query_positive.cpp:25]: Ah yes I did not pay enough attention to that

File: layers/gpu_validation/gpu_record.cpp [hunk 7]
```diff
@@ -533,7 +533,9 @@ void gpuav::Validator::PreCallRecordQueueSubmit(VkQueue queue, uint32_t submitCo
             PreRecordCommandBuffer(submit->pCommandBuffers[i]);
         }
     }
-    UpdateBDABuffer(app_buffer_device_addresses);
+    if (buffer_device_address_enabled) {
```
[the remark this answers is not in the document]
Comment (arno-lunarg) [reply layers/gpu_validation/gpu_record.cpp:536]: Yeah!
Also I had to disable the test that does a ray query in a vertex shader, it was crashing on mesa

File: tests/unit/gpu_av_ray_query_positive.cpp [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+    vk::CmdBindDescriptorSets(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_COMPUTE, pipeline.pipeline_layout_.handle(), 0, 1,
+                              &pipeline.descriptor_set_->set_, 0, nullptr);
+    vk::CmdDispatch(m_commandBuffer->handle(), 1, 1, 1);
+    m_commandBuffer->end();
+
+    m_commandBuffer->QueueCommandBuffer();
+    m_device->wait();
+}
+
+TEST_F(PositiveGpuAVRayQuery, DISABLED_RayQueryGraphicsBasic) {
```
Comment (spencer-lunarg) [inline tests/unit/gpu_av_ray_query_positive.cpp:282]: Add note what is wrong... I can take a look later on my Mesa machine, but nice to have that comment here

**Pull request merged**